src/components/common/DockerNetworkName.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react';
+import { AutoComplete, Form } from 'antd';
+import { usePrefixedTranslation } from 'hooks';
+import { useStoreActions } from 'store';
+
+type Status = '' | 'warning' | 'error' | undefined;
+
+interface Props {
+  name: string;
+  defaultValue?: string;
+  validateCallback?: (value: boolean) => void;
```

Comment:
Instead of bubbling up the validation status via props, you could use `rules` prop to specify a custom regex pattern. This would prevent the form from submitting, display the error message inline, and change the border color of the input field.
```ts
<Form.Item rules={[{ pattern: /^[a-zA-Z0-9][a-zA-Z0-9_.-]{1,63}$/, message: l('helpInvalid') }]}>
```